recon.py

```diff
@@ -0,0 +1,329 @@
+"""
+A Proposal for reconciliation graph and reconciliation interface.
+"""
+from abc import ABC  # Abstract Base Classes
+from typing import List, Dict
+from enum import Enum
+
+__all__ = ['MappingNode', 'EventType', 'Event', 'Cospeciation', 'Duplication', 'Transfer', 'Loss',
+           'TipTip', 'Reconciliation', 'ReconGraph']
+
+class MappingNode:
+    """
+    MappingNode is a node in the reconciliation graph that maps
+    a parasite to a host.
+    """
+    def __init__(self, parasite_vertex: str, host_vertex: str):
+        self._parasite = parasite_vertex
+        self._host = host_vertex
+
+    @property
+    def parasite(self) -> str:
```

Comment:
Is this the PEP standard, just the attribute name rather than get_attribute(name)?

Reply:
I haven't seen `@property` used in the style guide, but `@property` is built-in to python (see https://docs.python.org/3/library/functions.html#property). If we use `__getattr__`, we would have to write something like this to accomplish the same thing.:
```
if name == "host":
    return name.host
elif name == "parasite":
    return name.parasite
else:
    raise AttributeError
```
Which I think is more complicated than just using `@property`.